samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentFacadeQueries.java

```diff
@@ -724,8 +738,25 @@ public List<AssessmentFacade> getBasicInfoOfAllActiveAssessmentsByAgent(String o
 			if (questionSizeMap.get(a.getAssessmentBaseId()) != null) {
 				questionSize = ((Long) questionSizeMap.get(a.getAssessmentBaseId())).intValue();
 			}
-			AssessmentFacade f = new AssessmentFacade(a.getAssessmentBaseId(), a.getTitle(), a.getLastModifiedDate(), a.getStartDate(), a.getDueDate(), a.getReleaseTo(), releaseToGroups, lastModifiedBy, questionSize);
-			assessmentList.add(f);
+
+			if (releaseToGroups != null) {
+				Map<String, String> groupIdsMap = siteGroups.stream()
+					.collect(Collectors.toMap(Group::getId, Group::getId));
```

Comment:
should this happen in the assessments loop or outside of the loop?

Reply:
Changed outside the loop